google/resource_compute_subnetwork.go

```diff
@@ -10,285 +24,403 @@ import (
 		Create: resourceComputeSubnetworkCreate,
 		Read:   resourceComputeSubnetworkRead,
 		Update: resourceComputeSubnetworkUpdate,
 		Delete: resourceComputeSubnetworkDelete,
+
 		Importer: &schema.ResourceImporter{
-			State: resourceComputeSubnetworkImportState,
+			State: resourceComputeSubnetworkImport,
 		},
 
 		Timeouts: &schema.ResourceTimeout{
-			Create: schema.DefaultTimeout(6 * time.Minute),
-			Update: schema.DefaultTimeout(6 * time.Minute),
-			Delete: schema.DefaultTimeout(6 * time.Minute),
+			Create: schema.DefaultTimeout(360 * time.Second),
+			Update: schema.DefaultTimeout(360 * time.Second),
+			Delete: schema.DefaultTimeout(360 * time.Second),
 		},
+		CustomizeDiff: customdiff.All(
+			customdiff.ForceNewIfChange("ip_cidr_range", isShrinkageIpCidr),
+		),
 
 		Schema: map[string]*schema.Schema{
-			"ip_cidr_range": &schema.Schema{
+			"ip_cidr_range": {
 				Type:         schema.TypeString,
 				Required:     true,
 				ValidateFunc: validateIpCidrRange,
-				// ForceNew only if it shrinks the CIDR range, this is set in CustomizeDiff below.
 			},
-
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
-
-			"network": &schema.Schema{
+			"network": {
 				Type:             schema.TypeString,
 				Required:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
 			},
-
-			"description": &schema.Schema{
+			"description": {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
 			},
-
-			"fingerprint": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
-			},
-
-			"gateway_address": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
-			},
-
-			"project": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
-				ForceNew: true,
-			},
-
-			"region": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
-				ForceNew: true,
-			},
-
-			"private_ip_google_access": &schema.Schema{
+			"enable_flow_logs": {
 				Type:     schema.TypeBool,
 				Optional: true,
+				ForceNew: true,
 			},
-
-			"secondary_ip_range": &schema.Schema{
+			"secondary_ip_range": {
 				Type:     schema.TypeList,
 				Optional: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"range_name": &schema.Schema{
-							Type:         schema.TypeString,
-							Required:     true,
-							ValidateFunc: validateGCPName,
+						"ip_cidr_range": {
+							Type:     schema.TypeString,
+							Required: true,
 						},
-						"ip_cidr_range": &schema.Schema{
+						"range_name": {
 							Type:     schema.TypeString,
 							Required: true,
 						},
 					},
 				},
 			},
-
-			"enable_flow_logs": &schema.Schema{
+			"private_ip_google_access": {
 				Type:     schema.TypeBool,
 				Optional: true,
 			},
-
-			"self_link": &schema.Schema{
+			"region": {
```

Comment:
This should also be computed so it can be filled in if the user is using the provider-level region